### PR TITLE
Add PhotoOverlay, Camera, LookAt, AbstractView support

### LINKED ISSUE #19: Add PhotoOverlay support
https://developers.google.com/kml/documentation/kmlreference#photooverlay

### LINKED ISSUE #102: How to add LookAt
How can I add LookAt
I think it should be added to Document
d = kml.Document(ns, 'docid', 'doc name', 'doc description')

there should be something like

      <LookAt>
        <longitude>14.0</longitude>
        <latitude>44.0</latitude>
        <heading>0</heading>
        <tilt>0</tilt>
        <range>500</range>
      </LookAt>

If there's some replacement for LookAt that
provides same functionality, let me know

### PR DESCRIPTION
**Pull request by martimpassos**
Hi, still a bit of work to be done (and some useless functions I probably created in the heat of the moment), but I decided to open the PR to discuss a few points. Am comparing against `main` for easier merging, let me know if I should compare against `develop` instead. 
Fixes #102 
Fixes #19

### CODE REVIEW

File: fastkml/kml.py [hunk 1]
```diff
@@ -1990,7 +1990,7 @@ def heading(self):
 
     @heading.setter
     def heading(self, value):
-        if isinstance(value, (str, int, float)) and (0 <= float(value) <= 360):
+        if isinstance(value, (str, int, float)) and (-180 <= float(value) <= 180):
```
Comment (martimpassos) [inline fastkml/kml.py:2011]: Documentation states that 
```
Values range from 0 to 360 degrees.
```
but my sample file (created by Google Earth) has `-100` as value, so I'm guessing it's actually ±180

File: fastkml/kml.py [hunk 2]
```diff
@@ -1220,16 +1220,27 @@ def rotation(self, value):
         else:
             raise ValueError
 
-    def latLonBox(self, north, south, east, west, rotation=0):
-        # TODO: Check for bounds (0:+/-90 or 0:+/-180 degrees)
+    def LatLonBox(self, north, south, east, west, rotation=0):
```
Comment (martimpassos) [inline fastkml/kml.py:1239]: the project has mixed `camelCase` and `snake_case` names, not sure what I should default to. Specially in cases like ViewVolume and ImagePyramid that are attributes and do not have they're own class

File: fastkml/kml.py [hunk 3]
```diff
@@ -35,7 +35,7 @@
 # tried and tested dateutil here which is more stable. As a side effect
 # we can also parse non ISO compliant dateTimes
 import dateutil.parser
-from matplotlib.pyplot import phase_spectrum
+#from matplotlib.pyplot import phase_spectrum
```
Comment (martimpassos) [inline fastkml/kml.py:38]: can I remove this?

File: fastkml/kml.py [hunk 4]
```diff
@@ -963,9 +963,11 @@ def point(self):
         return self._point
 
     @point.setter
-    def point(self, point):
-        if isinstance(point.geometry, Point):
-            self._point = point
+    def point(self, value):
```
Comment (martimpassos) [inline fastkml/kml.py:982]: check that `point` is a `Point` instance?

File: fastkml/kml.py [hunk 5]
```diff
@@ -982,7 +982,7 @@ def shape(self, value):
         elif value is None:
             self._shape = None
         else:
-            raise ValueError
+            raise ValueError("Shape must be one of " "rectangle, cylinder, sphere")
```
Comment (martimpassos) [inline fastkml/kml.py:1001]: on line 1172 (GroundOverlay altitudeMode setter), the existing code sets to default value on invalid entries. Should I follow this or just raise an exception (which I think makes more sense)?

File: fastkml/kml.py (hunk 3 above)
Comment (cleder) [reply fastkml/kml.py:38]: yes sure, how the hell did this come in

File: fastkml/kml.py [hunk 6]
```diff
@@ -997,6 +997,77 @@ def ImagePyramid(self, tileSize, maxWidth, maxHeight, gridOrigin):
...
+                self.rightFov = rightFov.text
+            bottomFov = ViewVolume.find(f"{self.ns}bottomFov")
+            if bottomFov is not None:
+                self.bottomFov = bottomFov.text
+            topFov = ViewVolume.find(f"{self.ns}topFov")
+            if topFov is not None:
+                self.topFov = topFov.text
+            near = ViewVolume.find(f"{self.ns}near")
+            if near is not None:
+                self.near = near.text
```
Comment (martimpassos) [inline fastkml/kml.py:1065]: consider adopting the shorter
```
if attribute:
```
syntax throughout the project?

File: fastkml/kml.py (hunk 2 above)
Comment (cleder) [reply fastkml/kml.py:1239]: The camelCase names are legacy/technical debt, new code should use pep8 snake_case

File: fastkml/kml.py (hunk 1 above)
Comment (cleder) [reply fastkml/kml.py:2011]: Not sure if this should be checked after all: 361deg == 1deg

File: fastkml/kml.py (hunk 4 above)
Comment (cleder) [reply fastkml/kml.py:982]: not worth the effort, see #154

File: fastkml/kml.py (hunk 1 above)
Comment (martimpassos) [reply fastkml/kml.py:2011]: ok, just not sure if clients know how to handle 361deg, but I can test it

File: fastkml/kml.py (hunk 5 above)
Comment (cleder) [reply fastkml/kml.py:1001]: With type annotations, this check could be dropped entirely

File: fastkml/kml.py (hunk 6 above)
Comment (cleder) [reply fastkml/kml.py:1065]: If it works, I remember that Lxmls elements had a falsy value - but this is 10 years in the past

**Comment by martimpassos** [comment]
Sorry, the refactoring PRs are annoying, is there anyway I can avoid them?

**Comment by martimpassos** [comment]
Hello @cleder, do you have feedback on this PR? Please let me know

**Comment by dericke** [comment]
Hey @martimpassos, it looks as though there are a couple of easy fixes suggested [here](https://deepsource.io/gh/cleder/fastkml/run/ce1dd7fa-2879-47ff-905f-887b07594d90/python). Making the suggested changes should clear up one of the checks that's currently failing.

**Comment by martimpassos** [comment]
Thanks @dericke, I have addressed the issues in the code I introduced. The remaining ones are present in the currently published version.

**Pull request merged**